src/control-plane/backend/lambda/api/service/reporting.ts

```diff
@@ -0,0 +1,278 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { readFileSync } from 'fs';
+import { join } from 'path';
+import { AnalysisDefinition, ConflictException, DashboardVersionDefinition, QuickSight } from '@aws-sdk/client-quicksight';
+import { BatchExecuteStatementCommand, RedshiftDataClient } from '@aws-sdk/client-redshift-data';
+import { STSClient } from '@aws-sdk/client-sts';
+import { v4 as uuidv4 } from 'uuid';
+import {
+  createDataSet,
+  funnelVisualColumns,
+  getVisualDef,
+  getVisualRelatedDefs,
+  applyChangeToDashboard,
+  getDashboardDefinitionFromArn,
+  CreateDashboardResult,
+  sleep,
+  DashboardCreateParameters,
+  getCredentialsFromRole,
+} from './quicksight/reporting-utils';
+import { buildFunnelView } from './quicksight/sql-builder';
+import { awsAccountId, awsRegion } from '../common/constants';
+import { logger } from '../common/powertools';
+import { ApiFail, ApiSuccess } from '../common/types';
+
+const stsClient = new STSClient({ region: 'us-east-1' });
+const quickSight = new QuickSight({
+  region: awsRegion,
+});
+
+export class ReportingServ {
+
+  async createFunnelVisual(req: any, res: any, next: any) {
+    try {
+      logger.info(`request: ${JSON.stringify(req.body)}`);
+
+      const query = req.body;
+      const dashboardCreateParameters = query.dashboardCreateParameters as DashboardCreateParameters;
+
+      const credentials = await getCredentialsFromRole(stsClient, dashboardCreateParameters.dataApiRole!);
+      const redshiftDataClient = new RedshiftDataClient({
+        region: dashboardCreateParameters.redshiftRegion,
+        credentials: {
```

Comment:
Will it work if changing to `credentials,`?

Reply:
what is the meaning of changing to credentials?

Reply:
simplify the code